library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -309,7 +309,10 @@ String resolveTemplate(final String configTemplate, final String platformFilterT
 
     if (!stat_sinks_config.isEmpty()) {
       configBuilder.append("- &stats_sinks [");
-      configBuilder.append(String.join(",", stat_sinks_config));
+      configBuilder.append(stat_sinks_config.get(0));
+      for (int i = 1; i < stat_sinks_config.size(); i++) {
```

Comment:
that does not work with one stat sink?

Reply:
With just one stat sink, this for loop won't run at all, since the condition starts at `i = 1` and runs until `i < 1` which is never.